src/EnergyPlus/EvaporativeCoolers.cc

```diff
@@ -1045,35 +1041,29 @@ void InitEvapCooler(EnergyPlusData &state, int const EvapCoolNum)
     // Uses the status flags to trigger events.
 
     // Using/Aliasing
-    bool DoSetPointTest = state.dataHVACGlobal->DoSetPointTest;
-
-    // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-    bool localSetPointCheck(false);
-
-    auto &EvapCond(state.dataEvapCoolers->EvapCond);
+    auto &evapCond = state.dataEvapCoolers->EvapCond(EvapCoolNum);
 
     // Check that setpoint is active
-    if (!state.dataGlobal->SysSizingCalc && state.dataEvapCoolers->MySetPointCheckFlag && DoSetPointTest) {
+    if (!state.dataGlobal->SysSizingCalc && state.dataEvapCoolers->MySetPointCheckFlag && state.dataHVACGlobal->DoSetPointTest) {
         for (int EvapUnitNum = 1; EvapUnitNum <= state.dataEvapCoolers->NumEvapCool; ++EvapUnitNum) {
 
             // only check evap coolers that are supposed to have a control node
-            if ((EvapCond(EvapCoolNum).evapCoolerType != EvapCoolerType::IndirectRDDSpecial) &&
-                (EvapCond(EvapCoolNum).evapCoolerType != EvapCoolerType::DirectResearchSpecial))
+            if ((evapCond.evapCoolerType != EvapCoolerType::IndirectRDDSpecial) && (evapCond.evapCoolerType != EvapCoolerType::DirectResearchSpecial))
```

Comment:
This looks odd to me. This is a one time check in Init (managed using state.dataEvapCoolers->MySetPointCheckFlag) where all evap coolers are checked for a proper set point. However, inside this loop EvapCond(EvapCoolNum) is used. EvapCoolNum is the index to the evap cooler being simulated, not the loop index for this check of all evap coolers. This seems like a bug.

Reply:
Are you expecting this to cause diffs?  There are a few files with diffs that look like EvapCooler related files.

Reply:
I was expecting no diffs. I do expect that this will need to be changed evenutaly. I'll look at the diffs.